astroquery/xmatch/core.py

```diff
@@ -127,17 +127,22 @@ def _prepare_sending_table(self, cat_index, payload, kwargs, cat, colRA, colDec)
         catstr = 'cat{0}'.format(cat_index)
         if isinstance(cat, str):
             payload[catstr] = cat
-        elif isinstance(cat, Table):
-            # write the Table's content into a new, temporary CSV-file
-            # so that it can be pointed to via the `files` option
-            # file will be closed when garbage-collected
-            fp = StringIO()
-            cat.write(fp, format='ascii.csv')
-            fp.seek(0)
-            kwargs['files'] = {catstr: ('cat1.csv', fp.read())}
         else:
-            # assume it's a file-like object, support duck-typing
-            kwargs['files'] = {catstr: ('cat1.csv', cat.read())}
+            # create the dictionary of uploaded files
+            if "files" not in kwargs:
+                kwargs["files"] = {}
+            if isinstance(cat, Table):
+                # write the Table's content into a new, temporary CSV-file
+                # so that it can be pointed to via the `files` option
+                # file will be closed when garbage-collected
+
+                fp = StringIO()
+                cat.write(fp, format='ascii.csv')
+                fp.seek(0)
+                kwargs['files'].update({catstr: (f'cat{cat_index}.csv', fp.read())})
```

Comment:
I suppose you don't have to regenerate `cat{cat_index}` here, but reuse it from L127 as `f'{catstr}'`, however this is a ridiculous level of nitpicking, so I'll just go ahead and merge the fix instead :)

Reply:
Oh, yes it was better than what I did. Don't hesitate to change something if it happens again in an other PR, I won't be offended :slightly_smiling_face: